src/pages/TaskHistory.tsx

```diff
@@ -66,67 +69,94 @@ export const TaskHistory = () => {
+    const passingTests = toArray(queryParams[TestStatus.Passed]);
 
+    const failingTestFilters = failingTests.map((test) => ({
+      testName: test,
+      testStatus: TestStatus.Failed,
+    }));
+    const passingTestFilters = passingTests.map((test) => ({
+      testName: test,
+      testStatus: TestStatus.Passed,
+    }));
+    setHistoryTableFilters([...failingTestFilters, ...passingTestFilters]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [queryParams]);
   const queryParamsToDisplay = new Set([
     TestStatus.Failed,
     TestStatus.Passed,
     TestStatus.All,
   ]);
 
-  const selectedColumns = selectedBuildVariants?.length
-    ? buildVariantsForTaskName?.filter((bv) =>
-        selectedBuildVariants.includes(bv.buildVariant)
-      )
-    : buildVariantsForTaskName;
+  const selectedColumns = useMemo(
+    () =>
+      selectedBuildVariants?.length
+        ? buildVariantsForTaskName?.filter((bv) =>
+            selectedBuildVariants.includes(bv.buildVariant)
+          )
+        : buildVariantsForTaskName,
+    [buildVariantsForTaskName, selectedBuildVariants]
+  );
+
+  useEffect(() => {
+    console.log("Ran effect");
+    addColumns(toArray(selectedColumns).map((c) => c.buildVariant));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedColumns]);
   return (
     <PageWrapper>
       <CenterPage>
-        <HistoryTableProvider>
-          <PageHeader>
-            <H2>Task Name: {taskName}</H2>
-            <PageHeaderContent>
-              <HistoryTableTestSearch />
-              <BuildVariantSelector projectId={projectId} taskName={taskName} />
-            </PageHeaderContent>
-          </PageHeader>
-          <PaginationFilterWrapper>
-            <BadgeWrapper>
-              <FilterBadges queryParamsToDisplay={queryParamsToDisplay} />
-            </BadgeWrapper>
-            <ColumnPaginationButtons />
-          </PaginationFilterWrapper>
-          <TableContainer>
-            <ColumnHeaders loading={loading} columns={selectedColumns} />
-            <TableWrapper>
-              <HistoryTable
-                recentlyFetchedCommits={mainlineCommits}
-                loadMoreItems={() => {
-                  if (mainlineCommits) {
-                    setNextPageOrderNumber(mainlineCommits.nextPageOrderNumber);
-                  }
-                }}
-              >
-                {TaskHistoryRow}
-              </HistoryTable>
-            </TableWrapper>
-          </TableContainer>
-        </HistoryTableProvider>
+        <PageHeader>
+          <H2>Task Name: {taskName}</H2>
+          <PageHeaderContent>
+            <HistoryTableTestSearch />
+            <BuildVariantSelector projectId={projectId} taskName={taskName} />
+          </PageHeaderContent>
+        </PageHeader>
+        <PaginationFilterWrapper>
+          <BadgeWrapper>
+            <FilterBadges queryParamsToDisplay={queryParamsToDisplay} />
```

Comment:
It's a slightly weird UX that the "Clear all filters" button only appears once a string search has been added, but clicking it clears the selected build variants too. Maybe out of scope for this ticket, but would be nice to show this button once a build variant or task has been selected.

Reply:
Great catch. https://jira.mongodb.org/browse/EVG-15998